### PR TITLE
[fix](compaction) fix incorrect grouping of vertical compaction columns in tables only with key columns

### PR DESCRIPTION
**Pull request by luwei16**
When compaction deletes data through delete sign, it relies on this assumption: that is, the delete sign must be in the last column of the key group. When the unique table does not have a value column, vertical compaction triggers a corner case when grouping columns, and the above assumption cannot be guaranteed.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luwei16** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

File: be/src/olap/merger.cpp
Comment (gavinchou) [inline be/src/olap/merger.cpp:1]: should we also add a regression test for it?

Comment (zhannngchen) [reply be/src/olap/merger.cpp:1]: +1

File: be/src/olap/merger.cpp [hunk 1]
```diff
@@ -207,16 +207,25 @@ void Merger::vertical_split_columns(const TabletSchema& tablet_schema,
...
     for (uint32_t i = num_key_cols; i < total_cols; ++i) {
         if (i == sequence_col_idx || i == delete_sign_idx ||
             cluster_key_idxes.end() !=
                     std::find(cluster_key_idxes.begin(), cluster_key_idxes.end(), i)) {
             continue;
         }
-        if ((i - num_key_cols) % config::vertical_compaction_num_columns_per_group == 0) {
-            column_groups->emplace_back();
+
+        if (!value_columns.empty() &&
```
Comment (zhannngchen) [inline be/src/olap/merger.cpp:219]: For the unqiue key table,if there are only key columns in the schema, why does the code run here ? After all, besides the key column, there are only delete sign and sequence columns, which are skipped at line 213.

File: be/src/olap/merger.cpp
Comment (luwei16) [reply be/src/olap/merger.cpp:1]: done

**Comment by luwei16** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by luwei16** [comment]
run p0

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by zhannngchen** [comment]
run p0

**Comment by luwei16** [comment]
run p0

**Comment by luwei16** [comment]
run p0

**Comment by luwei16** [comment]
run buildall

**Comment by luwei16** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Pull request merged**